modules/ip/datapath/ip_ttl_exceeded.c

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2024 Christophe Fontaine
+
+#include <gr_datapath.h>
+#include <gr_graph.h>
+#include <gr_ip4_control.h>
+#include <gr_ip4_datapath.h>
+#include <gr_log.h>
+#include <gr_mbuf.h>
+
+#include <rte_common.h>
+#include <rte_graph_worker.h>
+#include <rte_icmp.h>
+#include <rte_ip.h>
+
+#define GR_IP_ICMP_TTL_EXCEEDED 11
+
+enum edges {
+	ICMP_OUTPUT = 0,
+	EDGE_COUNT,
+};
+
+static uint16_t
+ip_ttl_exceeded(struct rte_graph *graph, struct rte_node *node, void **objs, uint16_t nb_objs) {
+	rte_edge_t next = ICMP_OUTPUT;
+	struct rte_ipv4_hdr *ip;
+	struct ip_local_mbuf_data *ip_data;
+	struct rte_icmp_hdr *icmp;
+	struct rte_mbuf *mbuf;
+	const struct iface *input_iface;
+	uint16_t vrf_id;
+
+	for (uint16_t i = 0; i < nb_objs; i++) {
+		mbuf = objs[i];
+
+		ip = rte_pktmbuf_mtod(mbuf, struct rte_ipv4_hdr *);
+		// No need to check headroom
+		icmp = (struct rte_icmp_hdr *)
+			rte_pktmbuf_prepend(mbuf, sizeof(struct rte_icmp_hdr));
+
+		// Get the local router IP address from the input iface
+		input_iface = ip_output_mbuf_data(mbuf)->input_iface;
```

Comment:
It looks like you forgot to commit something else :) Nothing sets this input_iface value.

Reply:
indeed !